### ISSUE TITLE
Remove JAX-RS client dependencies from our client library

### ISSUE THREAD

**Issue by jmartisk** [smallrye/smallrye-graphql#436]
... and use pure HTTP

**Comment by t1** [comment]
If the client lives in an environment, where a jax-rs client is already available, e.g. in a JEE container like OpenLiberty, WildFly, etc., we should make sure not to include yet another library. Another alternative worth considering would be using the JDK 11 `HttpClient`, if that's available. We probably need a bunch of implementations to satisfy all of these requirements.

**Comment by t1** [comment]
@jmartisk : this issue is obsolete, now, isn't it?

**Comment by jmartisk** [comment]
I'd say we want to add support for using Vert.x HTTP client under the typesafe client library (instead of JAX-RS client), perhaps we should just rephrase this as such?

**Comment by t1** [comment]
Okay. But rebasing the Typesafe client on the Dynamic client (#725) would fix that, too, wouldn't it? My current assumption is that that's the way to go.

**Comment by jmartisk** [comment]
I think that is meant mainly for reusing some "core" stuff by the typesafe client, which can be done separately; so I'd personally prefer splitting it into two (if we actually do it in one PR in the end, that's fine)

The goal of #725 will be reusing some core stuff like the `Document` class, and #436 would be for creating a new variant of the typesafe client that uses Vert.x, while keeping the original JAX-RS one too? Does this sound ok?

**Comment by t1** [comment]
I'm more optimistic about a full rebase, i.e. making JAX-RS or a direct use of the Vert.x APIs obsolete. But it's okay for me to keep this issue open until we are sure how this actually turns out in the code.

**Comment by phillip-kruger** [comment]
Agree, the TypeSafe client should delegate all communication to the dynamic client

**Comment by jmartisk** [comment]
So we will drop the `client/implementation-jaxrs` module completely and only use vertx? (Sounds like less maintenance, which is good) And in that case I'd agree we can close this issue and do everything via #725

**Comment by phillip-kruger** [comment]
Yes I think that is the way to go.

**Comment by jmartisk** [comment]
Ok I'll close this then :)
